sunpy/net/dataretriever/attrs/adapt.py

```diff
@@ -0,0 +1,77 @@
+from sunpy.net.attr import SimpleAttr
+
+__all__ = ['ADAPTFileType', 'ADAPTLonType', 'ADAPTInputSource',
+           'ADAPTDataAssimilation', 'ADAPTResolution', 'ADAPTVersionYear', 'ADAPTVersionMonth',
+           'ADAPTEvolutionMode', 'ADAPTHelioData', 'ADAPTRealizations', 'ADAPTMagData']
+
+# Define a custom __dir__ to restrict tab-completion to __all__
+def __dir__():
+    return __all__
+
+
+class ADAPTFileType(SimpleAttr):
```

Comment:
Does the ADAPT API provide a range of possible values for any of these attributes?

Reply:
Hmmm, I think that's what the docstrings in this file are doing, is that not the case? All the possible values are also in the register_values method in the main file. How can I make this more compliant?

Reply:
It isn't about making it more compliant. For the VSO for example, we have a script that calls their API and it will update the values that are listed in "register_values". 

This avoids us having to do it manually and I was wondering if it was possible for ADAPT.

Reply:
I see... I'm not sure if they have an API, as this is just a web scraper client, but I can look into it.

Reply:
This is something, at least: https://gong.nso.edu/adapt/maps/adapt_filename_notes.txt

Reply:
Thanks.